packages/context/src/context.ts

```diff
@@ -90,7 +132,73 @@ export class Context {
     if (binding == null) return false;
     if (binding && binding.isLocked)
       throw new Error(`Cannot unbind key "${key}" of a locked binding`);
-    return this.registry.delete(key);
+    const found = this.registry.delete(key);
+    this.notifyListeners('unbind', binding);
```

Comment:
IIUC, `unbind` event is called even when the key was not found (`found === false`) and presumably no binding was deleted. Is that correct behavior? Wouldn't our users find it surprising?

Reply:
_reposting an older comment: https://github.com/strongloop/loopback-next/pull/2122/files#r245920613_

IIUC, `unbind` event is called even when the key was not found (`found === false`) and presumably no binding was deleted. Is that correct behavior? Wouldn't our users find it surprising?